### DISCUSSION TITLE
Toggle button to load script at the start

### DISCUSSION THREAD

**Discussion by gringow831** [tttapa/Control-Surface#705] [General]
Hi Pieter, 

Is there a way to use a toggle button to load up one part or another when powering up?

Let's say I have some banks and functions that are Channels 8 7 6 5 4 3 2 1  with the toggle pointing to the left and it will load by default.  If the toggle is to the right it will load 1 2 3 4 5 6 7 8 by default.  If it's possible how would I do that?

Thanks

**Comment by Veroledez** [comment]
Hola, en el void setup debes leer la palanca, y con la funcion .bankselect() selecccionar o activar los bancos, si tienes funciones como CCpotentiometer o demas, estas las puedes deshabilitar o habilitar con .enable() y .disable()

**Reply by gringow831** [reply]
Esto es lo que tengo escrito pero no logro que funcione
```
int buttonState = 0;
const int buttonAB = 21;

void setup() {
  Control_Surface.begin();
    pinMode(buttonAB, INPUT);
}

void loop() {
 buttonState = digitalRead(buttonAB);

if (buttonState == HIGH) {
 PBPotentiometer::enable(faderbanks);
PBPotentiometer::disable(faderbanks2);

  } else {
    PBPotentiometer::disable(faderbanks);
    PBPotentiometer::enable(faderbanks2);
  }
}

```

**Reply by gringow831** [reply]
This works perfectly for the LED pin 13 and it does what it's supposed to but not with the disable function. The Teensy flashes and there is no control
```
const int buttonAB = 21;
const int ledPin =  13; 
int buttonState = 0;

void setup() {
  RelativeCCSender::setMode(MACKIE_CONTROL_RELATIVE);
  Control_Surface.begin();
  pinMode(ledPin, OUTPUT);
  pinMode(buttonAB, INPUT);
}

void loop() { 
      Control_Surface.loop();
      buttonState = digitalRead(buttonAB);
      if (buttonState == HIGH) {  
      digitalWrite(ledPin, HIGH);
}  else {
      digitalWrite(ledPin, LOW);
  };
}
```

When I use this it doesn't. Not even the LED
```
void loop() { 
      Control_Surface.loop();
      buttonState = digitalRead(buttonAB);
if (buttonState == HIGH) {  
      digitalWrite(ledPin, HIGH);
      PBPotentiometer::enable(faderbanks);
      PBPotentiometer::disable(faderbanks2);
}  else {
      digitalWrite(ledPin, LOW); 
      PBPotentiometer::enable(faderbanks2);
      PBPotentiometer::disable(faderbanks);
  };
}
```

**Comment by Veroledez** [comment]
Utiliza la clase button de controlsurface,  te escribire una idea, no tengo el pc a mano para probar.

Button bot {2};

PBPotentiometer pot1 {A1, CHANNEL_1};
PBPotentiometer pot1 {A1, CHANNEL_1};

en void loop

If(boton.update()) {
If(boton.getstate()==falling){pot1.enable();
pot2.disable();}
if(boton.getstate()==rising){pot1.disable();
Pot2.enable()}
}

el codigo seguramente esta mal, es una mini guia  como para que experimentes.
revisa la biblioteca y ejemplos para que puedas organizar y compilar el codigo, no estoy en casa para brindarte un codigo funcional, espero y te sirva de guia, si surgue algo lo comentas, para ayudarte

**Reply by gringow831** [reply]
Eso se parece a mis primeros intentos.  Basandome en lo que me pasaste primero tengo que iniciar el botton bot en el setup. El pot2 no se desactiva nunca, ni en la primera fase del bot.update() asi que he probado desactivandolo en el setup. Ahi se desactiva obviamente. (He probado teniendolo las dos maneras)
En cualquiera de los casos siempre y cuando no tenga pot2.disable(); en el setup, los dos PBPotentiometers siguen funcionando y al apretar el Button bot my teensy deja de funcionar y el LED empieza a encenderse y apagarse en error
```
#include <Control_Surface.h>

USBMIDI_Interface midi;

Button bot {11};

PBPotentiometer pot1 {14, CHANNEL_1};
PBPotentiometer pot2 {15, CHANNEL_2};

void setup() {
  RelativeCCSender::setMode(MACKIE_CONTROL_RELATIVE);
  Control_Surface.begin();
  bot.begin(); //begin button
  }

void loop(){
Control_Surface.loop();
if(bot.update()) {
if(bot.getState()==Button::Falling){
  pot1.enable();
  pot2.disable();
 } else if (bot.getState()==Button::Rising){
  pot1.disable();
  pot2.enable();
}
}
}}
```

**Reply by Veroledez** [reply]
Es curioso, deberia funcionar, dejame intentarlo mañana que este en casa, haber que obtengo,  tal vez estamos pasando por alto algún detalle, mañana pruebo y te comparto mi experiencia.
Probare en un leonardo y esp32  que es lo que tengo

**Reply by gringow831** [reply]
A ver si tú ves el error porque he probado con algunas cosas diferentes y sigo teniendo el mismo resultado. Gracias!
